### PR TITLE
Follow ups on the custom marshaler PR

### PR DESCRIPTION
**Pull request by ArcturusZhang**
- Add a condition to avoid generating empty marshalJSON function when the struct only has READ-ONLY fields.
- A minor refactor to increase readability

### CODE REVIEW

File: src/Model/CodeModelGo.cs [hunk 1]
```diff
@@ -192,6 +192,10 @@ public IEnumerable<string> ModelImports
                 {
                     imports.UnionWith(CodeNamerGo.Instance.PageableImports);
                 }
+                if (ModelTypes.Any(mt => mt is CompositeTypeGo && mt.Cast<CompositeTypeGo>().NeedsCustomMarshalling))
+                {
+                    imports.Add(PrimaryTypeGo.GetImportLine("encoding/json"));
```
Comment (jhendrixMSFT) [inline src/Model/CodeModelGo.cs:197]: I found it, in CompositeTypeGo.cs:216.  Method AddImports(), you can move this to there.

Comment (ArcturusZhang) [reply src/Model/CodeModelGo.cs:197]: No wonder I just search `"encoding/json"` and found nothing...

**Comment by ArcturusZhang** [comment]
Hi @jhendrixMSFT I just made some changes, could you please have a look?

**Comment by fearthecowboy** [comment]
# 🤖 AutoRest automatic publish job 🤖
## success (version: 2.1.152)

**Pull request merged**